src/dotnet-install.sh

```diff
@@ -551,14 +551,20 @@ get_specific_product_version() {
         return 1
     fi
 
-    specific_product_version=$(curl -s --fail "$download_link")
-    if [ $? -ne 0 ]
+    if machine_has "curl"
```

Comment:
I'm not too strong on it, but should we better adapt and use download function instead? 
Some logic is duplicated.
get_latest_version_info() in similar case is using download function.

Reply:
Yes, we could benefit from an overall cleaning in the download methods. To not delay this fix however, I suggest we merge this small fix and leave refactoring to a different PR.

Reply:
could you please create the issue to track it please? we can drop it later if not needed

Reply:
Here is the issue: https://github.com/dotnet/install-scripts/issues/104